src/main/java/org/folio/persist/ItemRepository.java

```diff
@@ -81,15 +81,11 @@ public Future<List<Map<String, Object>>> getReindexItemRecords(String fromId, St
 
   public Future<List<Item>> updateItems(PgConnection conn, List<ItemPatch> items) {
     try {
-      if (items.size() > MAX_ENTITIES) {
-        var message = EXPECTED_A_MAXIMUM_RECORDS_TO_PREVENT_OUT_OF_MEMORY.formatted(MAX_ENTITIES, items.size());
-        var errors = new Errors().withErrors(List.of(
-          new Error()
-            .withMessage(message)
-            .withType("1")
-            .withCode("VALIDATION_ERROR")));
-        return Future.failedFuture(new ValidationException(errors));
+      Future<Void> validation = validateItemsSize(items);
+      if (validation.failed()) {
+        return validation.mapEmpty();
       }
```

Comment:
A Future should only be used for async calls.
This is a pure synchronous call.
How about this?
```
      if (items.size() > MAX_ENTITIES) {
        return Future.failedFuture(maxEntitiesException());
      }
```
```
  private static ValidationException maxEntitiesException() {
    …
  }
```